client/libs/ga-scroll-depth.js

```diff
@@ -0,0 +1,70 @@
+// Simplified/rewritten from https://github.com/leighmcculloch/gascrolldepth.js
+
+import { onWindowScrollThrottleDebounce$ } from '../js/utils/dom-events';
+
+const getDocumentHeight = () => {
+  return Math.max(
+    document.documentElement['scrollHeight'], document.body['scrollHeight'],
+    document.documentElement['offsetHeight'], document.body['offsetHeight'],
+    document.documentElement['clientHeight']
+  );
+};
+
+const getWindowHeight = () => {
+  return window.innerHeight || document.documentElement.clientHeight || document.body.clientHeight;
+};
+
+const getPageYOffset = () => {
+  return window.pageYOffset || (document.compatMode === 'CSS1Compat' ? document.documentElement.scrollTop : document.body.scrollTop);
+};
+
+const sendEvent = (action, label) => {
+  window.dataLayer.push({
+    'event': 'ScrollDistance',
+    'eventCategory': 'Scroll Depth',
+    'eventAction': action,
+    'eventLabel': label,
+    'eventValue': 1,
+    'eventNonInteraction': true
+  });
```

Comment:
Does GA automatically work with a `dataLayer`?
As far as I knew it was a concept, but you had to implement the logic of sending these events yourself?